HCS-1.md

```diff
@@ -35,15 +51,21 @@ A registry is employed to track uploaded data and their corresponding topic IDs
 
 - `p`: Protocol number
 - `op`: The operation type, here "deploy" indicates registering a new upload.
-- `t_id`: The topic ID where the image chunks are uploaded.
+- `t_id`: The topic ID where the file chunks are uploaded.
 - `to`: The wallet account ID associated with the upload.
 
-This entry is submitted to a predefined registry topic on Hedera, facilitating easy discovery and reference of uploaded images.
+This entry is submitted to a predefined registry topic on Hedera, facilitating easy discovery and reference of uploaded files.
 
 #### Retrieving and Reassembling
 
-To display the data, the application retrieves all chunk messages from the specified HCS topic. Chunks are sorted by their order index (`o`) and concatenated based on their content (`c`). The combined base64 string is then decoded back into binary image data for display.
+To display the data, the application retrieves all chunk messages from the specified HCS topic. Chunks are sorted by their order index (`o`) and concatenated based on their content (`c`). The combined base64 string is then decoded back into binary data for display.
+
+The complete steps are:
+1. Fetch all messages from the Topic ID, and sort them by their index `o`
+2. Concatenate the messages together, and decompress the file using a `zstd` decoding library.
```

Comment:
I think we need to include versatile compression of hex, base64, and zstd options for maximum flexibility for when we start reading other chain data in the standards. 

We could require the first message with order 'o': 0 to have the image type and compression type like base64 possibly?

Reply:
What are some examples of cross chain hangups that you would expect? 

Because my default thinking here is that unless we have a specific reason why zstd wouldn't work for all use cases we should keep things simple and support a single approach.

If we find in the future that it does break certain use cases, then we should have a new future proposal to support other compression algorithms and formats. Adding more options comes at additional compute and bandwidth costs because indexers, clients, and platforms must do additional work to support those options. Plus, if the current option saves the most compute and inscription costs it is for the benefit of all clients and platforms. In the event the current option doesn't work, we can always implement new options in the future.

As a compromise, what if we simply include in the memo the compression format used for future flexibility?

At 32 chars for SHA-256, we have plenty of room to fit in the algo. 

[hash]:[algo]:[encoding]

532eaabd9574880dbf76b9b8cc00832c20a6ec113d682299550d7a6e0f345e25:zstd:base64

Reply:
Nice, I like that approach with the topic id memo. I think I'm hanging myself up on reader problems and main standards scope. 

I was thinking if data was ported in a bridge from the witness data field of bitcoin (which is hexadecimal), to a topic id on hedera, then there's an opportunity for different data types to be present. Having the SHA-256 with the suffix the way you have it in the memo is great. And we can assume if the suffix is not present, std as default

Reply:
> Nice, I like that approach with the topic id memo. I think I'm hanging myself up on reader problems and main standards scope.
> 
> I was thinking if data was ported in a bridge from the witness data field of bitcoin (which is hexadecimal), to a topic id on hedera, then there's an opportunity for different data types to be present. Having the SHA-256 with the suffix the way you have it in the memo is great. And we can assume if the suffix is not present, zstd as default

Ah okay, the main issue with hexadecimal is that it takes up significantly more bytes versus base64, so storage costs are higher. I think converting back to base64 shouldn't be a huge hurdle. Maybe its up to us to create a slick UX to migrate bitcoin data without the drama, rather than put that burden of storing hexadecimal on the network (unless there's no other option)